### PR TITLE
Empty result returns error to use for 404

### PR DESCRIPTION
**Pull request by penguinpowernz**
Previously this was giving an undefined error object, and an array like `[{}]` as the objects causing strange behaviour in libraries that we using this connector as the datasource.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by jannyHou** [comment]
@slnode test please

**Comment by superkhau** [comment]
Can you add some tests to verify your changes and prevent regressions in the future?

**Comment by penguinpowernz** [comment]
Added a couple of tests and fixed linter issues, check it out

File: README.md [hunk 1]
```diff
@@ -47,5 +47,12 @@ var ds = new DataSource('redis');
...
-> 
+>
 > Existing data in database 0 may be deleted
+
+
+### Contributing
+
+When working on this library you should use the loopback eslint config by running:
+
+    $ sudo npm -g install eslint-config-loopback
```
Comment (superkhau) [inline README.md]: This is not needed, I see a bug in package.json where eslint + eslint-config-loopback is installed in deps instead of devDeps. I will fix it on master, please rebase and you should see the changes.

File: test/not-found.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+
```
Comment (superkhau) [inline test/not-found.test.js]: remove empty space

File: test/not-found.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+'use strict';
+
+var should = require('./init.js');
+
+var db, Model;
```
Comment (superkhau) [inline test/not-found.test.js]: move this in into the describe block before the "before" function. See https://github.com/strongloop/loopback-contributor-docs/blob/master/style-guide.md#layout-of-test-files

File: test/not-found.test.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    Model.all({where: {id: 999}}, function(err, objs) {
+      should.exist(err);
+      err.should.have.property('status');
+      err.status.should.equal(404);
+      done();
+    });
+  });
+});
+
+var id = 0;
```
Comment (superkhau) [inline test/not-found.test.js]: Here too https://github.com/strongloop/loopback-contributor-docs/blob/master/style-guide.md#layout-of-test-files

**Comment by superkhau** [comment]
Link to PR with ESLint fix https://github.com/strongloop/loopback-connector-redis/pull/33

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by penguinpowernz** [comment]
Sweet, how's it look now?

File: README.md [hunk 5]
```diff
@@ -47,5 +47,5 @@ var ds = new DataSource('redis');
 2. Run `npm test`
 
 > WARNING
-> 
+>
```
Comment (superkhau) [inline README.md]: Please add back the empty space to prevent the diff from showing unrelated changes. It's a nitpick anyways though, so it won't prevent landing if you don't fix this.

File: test/not-found.test.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+describe('when a model does exist', function() {
+  var should = require('./init.js');
+  var db, Model;
+  var id = 0;
+
+  before(createNewModel);
+
+  it('should not return an error', function(done) {
+    Model.all({where: {id: id}}, function(err, objs) {
+      should.not.exist(err);
```
Comment (superkhau) [inline test/not-found.test.js]: `if (err) return done(err)` for better stack tracing.

File: test/not-found.test.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      done();
+    });
+  });
+
+  function createNewModel(done) {
+    db = getSchema();
+    Model = db.define('Model', {name: String});
+
+    Model.create({name: 'horse'}, function(err, m) {
+      should.not.exist(err);
```
Comment (superkhau) [inline test/not-found.test.js]: ditto

**Comment by superkhau** [comment]
Can you rebase with master to get the latest changes? I also added few more comments, we're almost there. ;) Mostly LGTM now.

**Comment by penguinpowernz** [comment]
Oh man I fucked it up... what kind of sourcery is this rebasing...

**Comment by superkhau** [comment]
@penguinpowernz Do you still have your local? Feel free to submit another PR with the correct changes. Ping me in the new PR and I can give you a hand with rebasing. ;)
